source/blog/2017-02-13-automate-building-atomic-host.html.md

```diff
@@ -0,0 +1,117 @@
+
+The procedure and playbook below will enable you to create your own Atomic Host OSTree image. This is the first step in creating your own "distributions" of Atomic Host to install on your cloud servers. Note that it will install a bunch of requirements on your local server, as well as using system resources heavily. As such, you may want to run it on a development machine instead of your personal laptop.
+
+There is also Documentation on Project Atomic which describes about Compose own Tree and Build Atomic host: [http://www.projectatomic.io/docs/compose-your-own-tree](http://www.projectatomic.io/docs/compose-your-own-tree/).
+
+
+# Requirements
+* Fedora Atomic QCOW2 Image. The image can be downloaded from here: [Fedora-Atomic](https://getfedora.org/en/atomic/download/)
+* Make sure that Ansible is installed on your system: `$ sudo dnf install ansible`. Note that I am installing on my [Fedora-Workstation](https://getfedora.org/en/workstation/download).
+
+
+## Process
+Clone the Git repo on your working machine [Build-Atomic-Host](https://github.com/trishnaguha/build-atomic-host/)
+
+```
+$ git clone https://github.com/trishnaguha/build-atomic-host.git
+$ cd build-atomic-host
+```
+
+
+## Create VM from the QCOW2 Image
+The following creates VM from QCOW2 Image where username is `atomic-user` and password is `atomic`.
+Here `atomic-node` in the instance name.
+
+```
+$ sudo sh create-vm.sh atomic-node /path/to/fedora-atomic25.qcow2
+# For example: /var/lib/libvirt/images/Fedora-Atomic-25-20170131.0.x86_64.qcow2
+```
+
+
+## Install Requirements and Start HTTP Server
+The tree is made available via web server. The following playbook installs requirements, creates directory structure, initializes OSTree repository and starts the HTTP server.
+
+```
+$ ansible-playbook setup.yml --ask-sudo-pass
+```
+Use `ip addr` to check IP Address of the HTTP server.
+
+
+## Give OSTree a name and add HTTP Server IP Address
+
+Replace the variables given in [vars/atomic.yml](https://github.com/trishnaguha/build-atomic-host/tree/master/vars/atomic.yml) with OSTree name and HTTP Server IP Address.
+For Instance:
+
+```
+# Variables for Atomic host
+atomicname: my-atomic
+httpserver: 192.168.122.1
+```
+Here `my-atomic` is OSTree name and `192.168.122.1` is HTTP Server IP Address.
+
+
+## Run All-in-one Playbook
+The following playbook composes OSTree, performs SSH-setup and rebases on created Tree.
+
+```
+$ ansible-playbook main.yml --ask-sudo-pass
+```
+
+
+## Check IP Address of the Atomic instance
+The following command returns the IP Address of the running Atomic instance
+
+```
+$ sudo virsh domifaddr atomic-node
+```
+
+
+## Reboot
+Now SSH to the Atomic Host and reboot it so that it can reboot in to the created OSTree:
+
+```
+$ ssh atomic-user@<atomic-hostIP>
+$ sudo systemctl reboot
+```
+
+
+## Verify: SSH to the Atomic Host
+
+Wait for 10 minutes, You may want to go for a Coffee now.
```

Comment:
This is because to avoid `connection refused` issue after reboot. May be I should move this sentence above?